numpy/linalg/linalg.py

```diff
@@ -2180,11 +2180,14 @@ def lstsq(a, b, rcond="warn"):
     If `a` is square and of full rank, then `x` (but for round-off error)
     is the "exact" solution of the equation.
 
+    .. versionchanged:: 1.19
+       Can now operate on stacks of matrices
+
     Parameters
     ----------
-    a : (M, N) array_like
+    a : (..., M, N) array_like
```

Comment:
Hmm, the current docstring here is very old and could use improvement. Just array_like on one line, then a description of the shapes in the explanatory part and how they are treated.

Reply:
This style of docstring with the shape in the argument name is fairly prevalent throughout this module.

Reply:
Yep, could use improvement throughout. Not saying you should undertake the task in this PR.